src/lib.rs

```diff
@@ -12,6 +12,14 @@
 //! This is very similar to how `futures::stream::unfold` creates a `Stream`
 //! implementation from a seed value and a future-returning closure.
 //!
+//! # Error behaviour
+//!
+//! If any of the [`Sink`] methods produces an error, the sink transitions to
+//! a closed state. Invoking [`Sink::poll_ready`] or [`Sink::start_send`]
+//! again after an error has occured will cause a *panic*. Invoking
```

Comment:
If I am not mistaken, it will also panic when first closing the sink and then calling `poll_ready`, correct? The suggestion below should make that more clear.

```suggestion
//! on a sink in closed state will cause a *panic*. Invoking
```

Reply:
This is true. I did not think it needs to be made explicit that closing a sink and then trying to send items would lead to trouble but your suggestion covers all cases so thanks.